### ISSUE TITLE
Feature Suggestion: Support for modded item transfer

### ISSUE THREAD

**Issue by theboo** [MuradAkh/LittleLogistics#23]
Hoppers are a bit slow in today's modded minecraft. 

One suggestion i would have would be to have the tug and barge docks have an internal(even one slot) buffer that can transfer a stack at a time in/out of the barges/tugs depending on their configuration. So instead of the hoppers directly transferring in and out of the barges/tug, the docks do and you transfer out of the docks themselves. In a vanilla+ type pack you could still do this with hoppers above/below the dock blocks, but would also allow for other item transfer from other modded systems I.E. pipes, routers, etc. a bit faster.

I'm assuming with the future FE/Fluid transfer systems you will have to have some sort of compatibility with modded FE/fluid transfer, would make sense if items had the same :)

This is a super cool mod, excited to see where it goes!

**Comment by MuradAkh** [comment]
The fluid and energy loaders + vessels will be released should be released very very soon with beta.4. They should be much faster than hoppers. 

We will look into having faster item loader, thanks for the suggestion!

**Comment by theboo** [comment]
yeah, that's why i was thinking of having the docks themselves do the transfer, then you could just pipe in/out of them with whatever modded transfer you want :)

and will do! plan on doing a bunch more testing with this!

**Comment by MuradAkh** [comment]
One big advantage of not having docks do any transfering is that we just need one type of dock, and no need for loader/unloader and specific ones for fluids, energy etc. You should still be able to pipe stuff in quickly into a hopper (vanilla or possibly upcoming quick one from this mod). 

One current solution to slow item loading times is to simply add more barges. Chest barges and docks are cheap and there is no reason you need to fill them up, you can just distribute it across a bigger tug-trains, will look much cooler with a large tug train too!

**Comment by MuradAkh** [comment]
@theboo 
I wanted some of your input on the rapid hopper:

1. How expensive should the rapid hopper be? We are thinking about using some redstone and gold in addition to the vanilla hopper, but not sure how much of each resource. (how many dust and how many bars for each rapid hopper)

2. How fast should it be? We can have it be 4x or 8x as fast as the vanilla hopper.

**Comment by theboo** [comment]
Whoooops, idk why I didn't see the ping, so sorry for late reply... 

I think switching from iron to gold in the default hopper recipe would be totally fair for 8x transfer! 

Could put redstone in the 2 empty slots but I don't know that it needs it if it doesn't have any extra redstone functionality compared to the vanilla hopper

**Comment by MuradAkh** [comment]
rapid hopper has been added, but keeping this open in case we look further into supporting item transfer from other mods

**Comment by EDToaster** [comment]
this is currently being incubated via docking-capability-rework branch but unfortunately it won't make it in until 1.21 or later. Unfortunately I just don't have the time to fix all of the bugs yet :(
